examples/generative/stylegan.py

```diff
@@ -0,0 +1,736 @@
+"""
+Title: Face Generation and Mixing with StyleGAN
+Author: Soon-Yau Cheong
+Date created: 2021/06/27
+Last modified: 2021/06/27
+Description: Implementation of StyleGAN with custom train step.
+"""
+"""
+# StyleGAN
+
+
+This [StyleGAN](https://arxiv.org/abs/1812.04948) implementation is based on the book
+[Hands-on Image Generation with TensorFlow](https://www.amazon.com/dp/1838826785). The
+code from the book's [Github
+repository](https://github.com/PacktPublishing/Hands-On-Image-Generation-with-TensorFlow-2
+.0/tree/master/Chapter07) is refactored into custom train step to enable faster training
+time in non-eager mode. The key features of StyleGAN is to grow the resolution
+progressively and to incorporate style features in the generative process.
+"""
+
+"""
+## Import modules
+"""
+
+import os
+import random
+import math
+import numpy as np
+import matplotlib.pyplot as plt
+
+from enum import Enum
+from glob import glob
+from functools import partial
+
+import tensorflow as tf
+from tensorflow.keras.layers import Layer, Dense, Input, LeakyReLU, Reshape
+from tensorflow.keras.layers import UpSampling2D, AveragePooling2D, Flatten
+from tensorflow.keras.models import Sequential, load_model
+from tensorflow.keras.callbacks import ModelCheckpoint
+from tensorflow_addons.layers import InstanceNormalization
+
+
+"""
+## Prepare Dataset
+In this examples, we wiil use CelebA pre-resized to 128x128 resolution.
+"""
+
+CKPT_PATH = "./ckpt/64x64"
+DATA_PATH = "celeba_64x64/*.jpg"
+RESULT_PATH = "./results"
+os.makedirs(CKPT_PATH, exist_ok=True)
+os.makedirs(RESULT_PATH, exist_ok=True)
+
+"""shell
+curl -LO
+https://github.com/soon-yau/stylegan_keras/releases/download/keras_example/celeba_64x64.zip
+https://github.com/soon-yau/stylegan_keras/releases/download/keras_example/celeba_64x64.zip
+unzip celeba_64x64.zip -d .
+"""
+
+"""
+We use different batch size for different image resolution, with small batch size for
+high resolution so they fit into GPU memory. The training steps are scaled accordingly.
+"""
+
+BATCH_SIZE = {2: 16, 3: 16, 4: 16, 5: 16, 6: 16, 7: 8, 8: 4, 9: 4, 10: 4}
+TRAIN_STEP_RATIO = {k: BATCH_SIZE[2] / v for k, v in BATCH_SIZE.items()}
+
+
+def load(res, image_file):
+    image = tf.io.read_file(image_file)
+    image = tf.image.decode_jpeg(image)
+    image = tf.image.resize(
+        image, [res, res], method=tf.image.ResizeMethod.NEAREST_NEIGHBOR
+    )
+    image = tf.cast(image, tf.float32)
+    image = (image / 127.5) - 1
+    return image
+
+
+def create_dataset_(data_path, res):
+    res_log2 = int(np.log2(res))
+    BUFFER_SIZE = 100
```

Comment:
If it's a global variable it should be defined at the start of the file. But in this case you only use it once so you can inline the value.